src/frontend/src/optimizer/plan_node/batch_iceberg_scan.rs

```diff
@@ -29,22 +31,29 @@ use crate::error::Result;
 use crate::optimizer::plan_node::expr_visitable::ExprVisitable;
 use crate::optimizer::property::{Distribution, Order};
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Educe, Debug, Clone, PartialEq)]
+#[educe(Eq, Hash)]
 pub struct BatchIcebergScan {
     pub base: PlanBase<Batch>,
     pub core: generic::Source,
+    #[educe(Hash(ignore))]
+    pub predicate: IcebergPredicate,
```

Comment:
Hash and Eq are only required for streaming share plan. But down the road we may support batch share plan. In that case every single batch plan node using `Educe` needs to be audited.

Reply:
Not an issue after https://github.com/risingwavelabs/risingwave/pull/19228#discussion_r1834205783